frontend/src/app/components/login/login.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from 'src/app/services/auth.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  public state:string
+  public username:string;
+  public password:string;
+
+  constructor(private authService:AuthService, private router:Router) {
+    this.state = "unauthorized";
+    this.username="";
+    this.password="";
+  }
+
+  ngOnInit(): void {
+  }
+
+  onSubmit(){
+    this.authService.authenticate(this.username, this.password);
+    this.router.navigate(['/']);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ibmpariwesh/socgen-stockmarket-project/pullRequest?prid=8405405)
```suggestion
    this.router.navigate(["/"]);
```